pkg/sriov/sriov.go

```diff
@@ -169,16 +152,9 @@ func (s *sriovManager) ReleaseVF(conf *sriovtypes.NetConf, podifName string, cid
 			return fmt.Errorf("failed to rename link %s to host name %s: %q", podifName, conf.OrigVfState.HostIFName, err)
 		}
 
-		// reset effective MAC address
-		if conf.MAC != "" {
-			hwaddr, err := net.ParseMAC(conf.OrigVfState.EffectiveMAC)
-			if err != nil {
-				return fmt.Errorf("failed to parse original effective MAC address %s: %v", conf.OrigVfState.EffectiveMAC, err)
-			}
-
-			if err = s.nLink.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
-				return fmt.Errorf("failed to restore original effective netlink MAC address %s: %v", hwaddr, err)
-			}
+		err = utils.SetVFEffectiveMAC(s.nLink, conf.OrigVfState.HostIFName, conf.OrigVfState.EffectiveMAC)
```

Comment:
why is this bit not conditioned by `conf.MAC` as before ?

Reply:
note that `conf.OrigVfState.EffectiveMAC` is saved at L88 and is conditioned by `conf.MAC`

Reply:
we don't need it as we always have a mac address it can be the original one or the one saved in line 88

from main.go

```go
err = sm.FillOriginalVfInfo(netConf)
```

Reply:
FillOriginalVfInfo does not save effective MAC, see [1]

also in case `conf.MAC` was not specified i dont think we need to restore it as we did not set it to begin with.
my POV here is why add un-needed steps that can theoretically fail.


[1] https://github.com/k8snetworkplumbingwg/sriov-cni/blob/77c66ac82231a327f147fe7d29bea9c70ce06ffa/pkg/types/types.go#L22

Reply:
you are right nice catch!